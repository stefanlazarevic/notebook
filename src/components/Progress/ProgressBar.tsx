import React, { useMemo } from "react";

import "./ProgressBar.css";

import { ProgressBarProps, ProgressBarPropTypes } from "./ProgressBarProps";

function ProgressBar(props: ProgressBarProps) {
	const className = useMemo(() => `ProgressBar ${props.className}`.trim(), [
		props.className,
	]);

	const style = useMemo(() => {
		if (!props.value || props.value < 0) {
			return { width: "0%" };
		}

		if (props.value > 100) {
			return { width: "100%" };
		}

		return { width: `${props.value}%` };
	}, [props.value]);

	return (
		<div
			id={props.id}
			tabIndex={0}
			data-testid={props.testid}
			className={className}
			role="progressbar"
			aria-valuemin={0}
			aria-valuemax={100}
			aria-valuenow={props.value}
			aria-busy={props["aria-busy"]}
			aria-describedby={props["aria-describedby"]}
			aria-valuetext={props["aria-valuetext"] || `${props.value}%`}
		>
			<div className="Progress" style={style} aria-hidden={true} />
		</div>
	);
}

ProgressBar.propTypes = ProgressBarPropTypes;

ProgressBar.defaultProps = {
	value: 0,
	label: false,
};
ProgressBar.displayName = "ProgressBar";

export default ProgressBar;
