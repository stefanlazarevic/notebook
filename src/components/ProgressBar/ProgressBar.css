.ProgressBar {
    box-sizing: border-box;

    position: relative;

    width: 100%;

    min-height: 2px;

    height: 20px;

    background-color: #eee;

    border: 1px solid #eee;
}

.ProgressBar:focus {
    outline: 2px solid #a06b9a;

    outline-offset: 2px;
}

.ProgressBar::after {
    box-sizing: border-box;

    content: attr(aria-valuetext);

    position: absolute;

    top: 50%;

    left: 10px;

    transform: translateY(-50%);

    color: #fff;
}

.ProgressBar > .Progress {
    box-sizing: border-box;

    position: absolute;

    top: 0;

    left: 0;

    height: 100%;

    background-color: #a06b9a;
}
