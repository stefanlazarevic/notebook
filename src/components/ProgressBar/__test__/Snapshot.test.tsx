import React from "react";
import { render, cleanup } from "@testing-library/react";
import ProgressBar from "../index";

afterEach(cleanup);

it("Подразумеванa својства се подударају са последњим снимком.", () => {
	const { asFragment } = render(<ProgressBar />);

	expect(asFragment()).toMatchSnapshot();
});

it("Сва подешена својства се подударају са последњим снимком.", () => {
	const { asFragment } = render(
		<ProgressBar
			id="id"
			testid="test-id"
			className="TestProgressBar"
			value={40}
			aria-describedby="test-label"
			aria-busy={false}
			aria-valuetext="Loading..."
		/>
	);

	expect(asFragment()).toMatchSnapshot();
});
