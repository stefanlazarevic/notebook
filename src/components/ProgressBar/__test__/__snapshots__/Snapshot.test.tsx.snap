// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Подразумеванa својства се подударају са последњим снимком. 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="0"
    aria-valuetext="0%"
    class="ProgressBar"
    role="progressbar"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="Progress"
      role="presentation"
      style="width: 0%;"
    />
  </div>
</DocumentFragment>
`;

exports[`Сва подешена својства се подударају са последњим снимком. 1`] = `
<DocumentFragment>
  <div
    aria-busy="false"
    aria-describedby="test-label"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="40"
    aria-valuetext="Loading..."
    class="ProgressBar TestProgressBar"
    data-testid="test-id"
    id="id"
    role="progressbar"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="Progress"
      role="presentation"
      style="width: 40%;"
    />
  </div>
</DocumentFragment>
`;
